geti_sdk/data_models/credit_system.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (C) 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+"""
+Credit system-related entities
+"""
+
+from copy import deepcopy
+from pprint import pformat
+from typing import Any, Dict, Optional
+
+import attr
+
+from geti_sdk.data_models.enums import SubscriptionStatus
+from geti_sdk.data_models.utils import (
+    attr_value_serializer,
+    deidentify,
+    remove_null_fields,
+    str_to_datetime,
+    str_to_enum_converter,
+)
+
+
+@attr.define
+class CreditBalance:
+    """
+    Representation of the Credit Balance in Intel Geti
+    """
+
+    incoming: int
+    available: int
+    blocked: Optional[int] = None
+
+
+@attr.define
+class CreditAccount:
+    """
+    Representation of the Credit Account in Intel Geti
+    """
+
+    _identifier_fields = [
+        "id",
+        "organization_id",
+    ]
+    id: str
+    organization_id: str
+    name: str
+    balance: CreditBalance
+    created: str = attr.field(converter=str_to_datetime)
+    updated: str = attr.field(converter=str_to_datetime)
+    expires: Optional[str] = attr.field(
+        default=None, converter=str_to_datetime
+    )  # renewable account doesn't have an expiration timestamp by default
+    renewal_day_of_month: Optional[str] = attr.field(
+        default=None, converter=str_to_datetime
+    )
```

Comment:
`renewal_day_of_month` is just an `int` number (e.g., for _July, 31_ it'd be `31`)

Reply:
we convert it to `datetime.datetime` and `str` is the correct typing accepted by attr in this case